packages/datastore/datastore/src/replication/mutations/QueueUpdateProcessor.ts

```diff
@@ -0,0 +1,138 @@
+import { ModelMap } from "./MutationsQueue";
+import { MutationRequest } from "./MutationRequest";
+import { LocalStorage, CRUDEvents } from "../../storage";
+import { Model } from "../../Model";
+import traverse from "traverse";
+
+/**
+ * Swaps all client ids for server ids in queue
+ * and updates corresponding item in LocalStorage
+ */
+export class QueueUpdateProcessor {
```

Comment:
I think we will need to also handle addingNew item to queue from here to being able to add mutexes (separate PR/issue). Thx.